handlers/common.go

```diff
@@ -0,0 +1,27 @@
+package handlers
+
+import (
+	"context"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
+)
+
+func getReleaseDate(ctx context.Context, dc DatasetClient, userAuthToken, collectionID, datasetID, edition, versionID string) (string, error) {
+	var vErr error
+	var version, initialVersion dataset.Version
+
+	version, vErr = dc.GetVersion(ctx, userAuthToken, "", "", collectionID, datasetID, edition, versionID)
+	if vErr != nil {
+		return "", vErr
+	}
+
+	if version.Version != 1 {
+		initialVersion, vErr = dc.GetVersion(ctx, userAuthToken, "", "", collectionID, datasetID, edition, "1")
+		if vErr != nil {
+			return "", vErr
+		}
+		return initialVersion.ReleaseDate, nil
+	} else {
+		return version.ReleaseDate, nil
+	}
+}
```

Comment:
There is `handlers.go` file that has the common `setStatusCode` function so I think the functions should be together but probably in a file called `common.go`

Reply:
I've combined the files as `common.go`